controllers/jujucluster_controller.go

```diff
@@ -319,24 +325,33 @@ func (r *JujuClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{RequeueAfter: requeueTime}, nil
 	}
 
-	// Check if credential has been created yet
-	credExistsInput := juju.CredentialExistsInput{
-		CredentialName: jujuCluster.Name,
-		CloudName:      jujuCluster.Name,
-	}
-	credentialExists, err := jujuClient.Credentials.CredentialExists(ctx, credExistsInput)
-	if err != nil {
-		log.Error(err, "failed to query existing credentials")
-		return ctrl.Result{}, err
-	}
-	if !credentialExists {
-		log.Info("credential not found, creating it now")
-		err = createCredential(ctx, cloudSecret, jujuCluster, jujuClient)
+	if len(jujuCluster.Spec.Cloud.AuthTypes) > 0 {
+		// Check if credential has been created yet
+		credExistsInput := juju.CredentialExistsInput{
+			CredentialName: jujuCluster.Name,
+			CloudName:      jujuCluster.Name,
```

Comment:
I would expect this to use the cloud name from the spec:

```suggestion
			CredentialName: jujuCluster.Name,
			CloudName:      jujuCluster.Spec.Cloud.Name,
```

I feel like the CredentialName should have a different value, too, but I'm not quite sure what it should be. It looks like the createCredential function uses a name that's contained in the credential secret data, so I would think we should use the same name here, right?

Reply:
alright I have fixed the cloud name to use the value from the spec, and the credential name is obtained from the secret data